// bakong-khqr/src/constant/KHQRSubtag.js
const { emv } = require(".");

module.exports = {
    input: [
        {
            tag: "29",
            data: {
                bakongAccountID: null,
                accountInformation: null,
            },
        },
        {
            tag: "30",
            data: {
                bakongAccountID: null,
                merchantID: null,
                acquiringBank: null,
            },
        },
        {
            tag: "62",
            data: {
                billNumber: null,
                mobileNumber: null,
                storeLabel: null,
                terminalLabel: null,
            },
        },
    ],
    compare: [
        {
            tag: "29",
            subTag: emv.BAKONG_ACCOUNT_IDENTIFIER,
            name: "bakongAccountID",
        },
        {
            tag: "29",
            subTag: emv.MERCHANT_ACCOUNT_INFORMATION_MERCHANT_ID,
            name: "merchantID",
        },
        {
            tag: "29",
            subTag: emv.MERCHANT_ACCOUNT_INFORMATION_ACQUIRING_BANK,
            name: "acquiringBank",
        },
        {
            tag: "62",
            subTag: emv.BILLNUMBER_TAG,
            name: "billNumber",
        },
        {
            tag: "62",
            subTag: emv.ADDITIONAL_DATA_FIELD_MOBILE_NUMBER,
            name: "mobileNumber",
        },
        {
            tag: "62",
            subTag: emv.STORELABEL_TAG,
            name: "storeLabel",
        },
        {
            tag: "62",
            subTag: emv.TERMINAL_TAG,
            name: "terminalLabel",
        },
    ],
};
